backend/session/direct.c

```diff
@@ -245,6 +245,7 @@ static struct wlr_session *direct_session_create(struct wl_display *disp) {
 		wlr_log_errno(WLR_ERROR, "Allocation failed");
 		return NULL;
 	}
+	wl_signal_init(&session->base.session_signal);
```

Comment:
Would it make sense to move all the bits initializing the session into `wlr_session_init`?

Reply:
I had thought about that, but was unsure. I added one now, but:

1. wlr_session_init seems misleading when wlr_session_create also exists.
2. Moving udev in there would require moving teardown as well, for when backends are tried in series.

Reply:
>wlr_session_init seems misleading when wlr_session_create also exists.

Actually it doesn't seem like `session_impl` is exposed in a public header. That means session implementations can only exist in wlroots (third-party impls aren't supported). Can we move `wlr_session_init` to a private header? (And remove the `wlr_` prefix)

Reply:
Done. `session_impl` is actually exposed as an unstable feature in `include/wlr/backend/sesion/interface.h`, but we can always make `session_init` public later if there's a need for it.

Reply:
Hmm. We should really prefix it with `wlr_`, but that's unrelated to this PR.